### DISCUSSION TITLE
Set maximum packet size for sftp upload

### DISCUSSION THREAD

**Discussion by IvanMoscovic** [hierynomus/sshj#798] [Q&A]
Hello, may I ask if there is any possibility to set maximum packet size for sftp upload? We are having problems with bad network connection in some places, where we need to upload some bigger files and from what I found in the code, packet size is determined from MTU (max transfer unit) on the device. Is there any chance how to set it for net.schmizz.sshj.sftp.SftpClient?

**Comment by exceptionfactory** [comment]
The [Connection.setMaxPacketSize()](https://github.com/hierynomus/sshj/blob/1d8eaa7ce29c9d08173d1ccf5b8dcf6b3871e356/src/main/java/net/schmizz/sshj/connection/Connection.java#L126) method supports configuring the local maximum packet size in bytes. The default value is [32 KB](https://github.com/hierynomus/sshj/blob/1d8eaa7ce29c9d08173d1ccf5b8dcf6b3871e356/src/main/java/net/schmizz/sshj/connection/ConnectionImpl.java#L55). The `Connection` can be retrieved using `SSHClient.getConnection()`, which should be called and configured prior to creating a new `SFTPClient`.
